pkg/mark/macro/macro.go

```diff
@@ -134,13 +134,36 @@ func ExtractMacros(
 				macro Macro
 			)
 
-			macro.Template, err = includes.LoadTemplate(base, template, templates)
-			if err != nil {
-				err = karma.Format(err, "unable to load template")
+			if strings.HasPrefix(template, "#") {
+				cfg := map[string]interface{}{}
 
-				return nil
+				err = yaml.Unmarshal([]byte(config), &cfg)
+				if err != nil {
+					err = karma.Format(
+						err,
+						"unable to unmarshal macros config template",
+					)
+					return nil
+				}
+				body, _ := cfg[template[1:]].(string)
```

Comment:
Will it always (like in 100% cases) go well? It doesn't feel too safe actually. Let's check that there is such a field in cfg first with comma-ok pattern?

Reply:
As far as I understand, yes.

template is a string of at least 1 character, which will be '#' due to https://github.com/Skeeve/mark/blob/b7a30d6cf4e3fe0984cea44b4101e512b247fdfb/pkg/mark/macro/macro.go#L137

Also body will either be the inline macro's content or an empty string as, as far as I know, in case of a not-present map element, the result will be the empty string here:

https://github.com/Skeeve/mark/blob/b7a30d6cf4e3fe0984cea44b4101e512b247fdfb/pkg/mark/macro/macro.go#L148

So while it might produce unexpected results, it shouldn't crash.

But yes… An appropriate error message might be more helpful.